File: ui-bookmarks/src/main/java/l/files/ui/bookmarks/RemoveBookmark.java
package l.files.ui.bookmarks;

import android.support.v7.view.ActionMode;
import android.view.Menu;
import android.view.MenuItem;

import l.files.bookmarks.BookmarkManager;
import l.files.fs.File;
import l.files.ui.base.selection.Selection;
import l.files.ui.base.view.ActionModeItem;

import static android.view.Menu.NONE;
import static l.files.base.Objects.requireNonNull;

final class RemoveBookmark extends ActionModeItem {

    private final BookmarkManager bookmarks;
    private final Selection<File, ?> selections;

    RemoveBookmark(Selection<File, ?> selection, BookmarkManager bookmarks) {
        super(R.id.delete_selected_bookmarks);
        this.selections = requireNonNull(selection);
        this.bookmarks = requireNonNull(bookmarks);
    }

    @Override
    public boolean onCreateActionMode(ActionMode mode, Menu menu) {
        super.onCreateActionMode(mode, menu);
        menu.add(NONE, id(), NONE, R.string.remove)
                .setIcon(R.drawable.ic_remove_circle_outline_white_24dp);
        return true;
    }

    @Override
    protected void onItemSelected(ActionMode mode, MenuItem item) {
        bookmarks.removeBookmarks(selections.keys());
        mode.finish();
    }

}

File: ui-bookmarks/src/main/java/l/files/ui/bookmarks/RemoveBookmark.kt
package l.files.ui.bookmarks

import android.view.Menu
import android.view.Menu.NONE
import android.view.MenuItem
import androidx.appcompat.view.ActionMode
import l.files.base.lifecycle.SetLiveData
import l.files.ui.base.selection.Selection
import l.files.ui.base.view.ActionModeItem
import java.nio.file.Path

internal class RemoveBookmark(
  private val selections: Selection<Path, *>,
  private val bookmarks: SetLiveData<Path>
) : ActionModeItem(R.id.delete_selected_bookmarks) {

  override fun onCreateActionMode(mode: ActionMode, menu: Menu): Boolean {
    super.onCreateActionMode(mode, menu)
    menu.add(NONE, id(), NONE, R.string.remove)
      .setIcon(R.drawable.ic_remove_circle_outline_white_24dp)
    return true
  }

  public override fun onItemSelected(mode: ActionMode, item: MenuItem) {
    bookmarks.removeAll(selections.keys())
    mode.finish()
  }

}
